src/view_models/AccessGroupViewModel.js

```diff
@@ -0,0 +1,221 @@
+    this.requires_tos = false;
+    this.accepted_tos = false;
+    this.state = '';
+    this.error = false;
+    this.processData(data);
+  }
+
+  processData(data) {
+    try {
+      this.group_name = data.group_name;
+      this.requires_tos = data.terms;
+      this.user_uuid = data.user_uuid;
+      if ('state' in data) {
+        this.state = data.state;
+      }
+    } catch (e) {
+      this.error = e.message;
+      console.error('GroupInvitation error: ', e.message);
+      throw new Error(e.message);
+    }
+  }
+}
+
+export class HistoricalEventViewModel {
+  constructor(data) {
+    this.error = false;
+    try {
+      this.privilege = data.privilege;
+      this.date_added = data.date_added;
+    } catch (e) {
+      this.error = e.message;
+      console.error('TypeValue error: ', e.message);
+      throw new Error(e.message);
+    }
+  }
+}
+
+export class TypeValueViewModel {
+  constructor(data) {
+    this.error = false;
+    try {
+      this.type = data.type;
+      this.value = data.value;
+    } catch (e) {
+      this.error = e.message;
+      console.error('TypeValue error: ', e.message);
+      throw new Error(e.message);
+    }
+  }
+}
+
+const defaultUserName = 'Anonymous User';
+
+export class AbbDisplayMemberViewModel {
+  constructor(data) {
+    this.uuid = '';
+    this.name = '';
+    this.username = '';
+    this.email = '';
+    this.error = false;
+    this.processData(data);
+  }
+
+  processData(data) {
+    try {
+      this.uuid = data.user_uuid;
+      this.name =
+        !data.first_name && !data.last_name
+          ? defaultUserName
+          : `${data.first_name} ${data.last_name}`;
+      this.username = data.username;
+      this.email = data.email;
+    } catch (e) {
+      this.error = e.message;
+      console.error('Abbreviated Display Member error: ', e.message);
+    }
+  }
+
+  isAnonymous() {
+    return this.username === 'anonymous';
+  }
+}
+export const MEMBER_IDEX = {
+  Curator: 0,
+  Member: 1,
+};
+export const DISPLAY_MEMBER_ROLES = ['Curator', 'Member'];
+export class DisplayMemberViewModel {
+  constructor(data) {
+    this.uuid = '';
+    this.picture = '';
+    this.name = '';
+    this.username = '';
+    this.email = '';
+    this.isStaff = false;
+    this.since = '';
+    this.expiration = 0;
+    [, , this.role] = DISPLAY_MEMBER_ROLES;
+    this.added_by = {};
+    this.error = false;
+    if (!data) {
+      this.error = 'DisplayMemberViewModel data is empty';
+      return;
+    }
+    this.processData(data);
+  }
+
+  static fromUserData(user) {
+    const member = new DisplayMemberViewModel();
+    member.uuid = user.user_uuid;
+    member.name = `${user.first_name} ${user.last_name}`;
+    member.picture = user.picture;
+    member.email = user.email;
+    member.username = user.username;
+    member.error = false;
+    return member;
+  }
+
+  processData(data) {
+    try {
+      this.uuid = data.user_uuid;
+      this.picture = data.picture;
+      this.name =
```

Comment:
Maybe we can rename this to `displayName` and use fallbacks:
_firstName / lastName → username → defaultUserName_

Reply:
Solid idea...